straxen/analyses/holoviews_waveform_display.py

```diff
@@ -187,26 +195,63 @@ def _hvdisp_plot_records_2d(records,
     return shader, records, time_stream
 
 
-def plot_record_polygons(record_points, width=1.1, **kwargs):
+def plot_record_polygons(record_points,
+                         center_time=True,
+                         scaling=10**-3,):
     """
     Plots record hv.Points as polygons for record matrix.
 
     :param record_points: Holoviews Points generated with
         _records_to_points.
-    :param width: Length of the record in µs.
-    :param kwargs: Keyword arguments applied to hv.Polygons options.
+    :param center_time: If true treats specified times as center times.
+    :param scaling: Scale times scale by this factor. E.g. if times
+        should be in µs use 10**-3.
     :returns: hv.Polygons
     """
     import holoviews as hv
 
-    data = [{('x', 'y'): rectangle(t, ch, width=width), 'area': a} for t, ch, a in
-            record_points.data[['time', 'channel', 'area']].values]
-    polys = hv.Polygons(data, vdims='area')
-    polys = polys.opts(color='level', aspect=4, responsive='width', line_width=0, **kwargs, cmap='viridis')
+    df = record_points.data.copy()
+    if not center_time:
+        df.loc[:, 'time'] = df.loc[:, 'time'] + (df.loc[:, 'length'] / 2 * df.loc[:, 'dt'])
+
+    data = _make_polys(df,
+                       scaling=scaling)
+    polys = hv.Polygons(data,
+                        kdims=['time', 'channel'],
+                        vdims=list(data[0].keys())[1:]).opts(color='area',
+                                                             aspect=4,
+                                                             responsive='width',
+                                                             line_width=0,
+                                                             cmap='viridis',
+                                                             )
+
     return polys
 
 
-def rectangle(time=0, channel=0, width=1.1, height=1):
+def _make_polys(df, scaling=1):
+    """
+    Function which converts the hitlet/record points into polygons
+    for the hitlet/record matrix.
+    """
+    res = []
+    for row_i in range(len(df)):
```

Comment:
Seems a bit strange to iterate over rows, not crucial i guess since number of rows should be small but why not have pandas do the iteration for you?

Reply:
Okay this function will become obsolete once I refactored things.